c10/core/DispatchKeySet.cpp

```diff
@@ -12,10 +12,46 @@ constexpr DispatchKeySet autograd_dispatch_keyset = DispatchKeySet({
   DispatchKey::AutogradOther,
 });
 
+constexpr DispatchKeySet math_dispatch_keyset = DispatchKeySet({
+  DispatchKey::CPU,
+  DispatchKey::CUDA,
+  DispatchKey::HIP,
+  DispatchKey::FPGA,
+  DispatchKey::MSNPU,
+  DispatchKey::XLA,
+  DispatchKey::Vulkan,
+  DispatchKey::MKLDNN,
+  DispatchKey::OpenGL,
+  DispatchKey::OpenCL,
+  DispatchKey::IDEEP,
+  DispatchKey::QuantizedCPU,
+  DispatchKey::QuantizedCUDA,
+  DispatchKey::ComplexCPU,
+  DispatchKey::ComplexCUDA,
+  DispatchKey::CustomRNGKeyId,
+  DispatchKey::MkldnnCPU,
+  DispatchKey::SparseCPU,
+  DispatchKey::SparseCUDA,
+  DispatchKey::SparseHIP,
+  DispatchKey::PrivateUse1,
+  DispatchKey::PrivateUse2,
+  DispatchKey::PrivateUse3,
+  DispatchKey::AutogradCPU,
+  DispatchKey::AutogradCUDA,
+  DispatchKey::AutogradXLA,
+  DispatchKey::AutogradPrivateUse1,
+  DispatchKey::AutogradPrivateUse2,
+  DispatchKey::AutogradPrivateUse3,
+  DispatchKey::AutogradOther,
+});
+
+
 DispatchKeySet getRuntimeDispatchKeySet(DispatchKey t) {
   switch (t) {
     case DispatchKey::Autograd:
       return autograd_dispatch_keyset;
+    case DispatchKey::Math:
```

Comment:
Same DRY note for this function - we shouldn't be repeating this mapping if we can help it, instead ideally `getRuntimeDispatchKeySet(t)` would be based on `getRuntimeDispatchKeys(t)` which "owns" the mapping.